warehouse/query-core/src/main/java/datawave/query/tables/BaseScannerSession.java

```diff
@@ -0,0 +1,14 @@
+package datawave.query.tables;
+
+import com.google.common.util.concurrent.AbstractExecutionThreadService;
+
+import java.util.Iterator;
+
+/**
+ * This will handles running a scan against a set of ranges. The actual scan is performed in a separate thread which places the results in a result queue. The
```

Comment:
AI did not write this line.

Reply:
Indeed. AI probably would have made a cogent comment, ha. Thanks for pointing this out